src/Discord.Net.Commands/Results/PreconditionGroupResult.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Discord.Commands
+{
+    [DebuggerDisplay(@"{DebuggerDisplay,nq}")]
+    public struct PreconditionGroupResult : IResult
+    {
+        public CommandError? Error { get; }
+
+        public string ErrorReason { get; }
+
+        public bool IsSuccess => !Error.HasValue;
+
+        public IEnumerable<PreconditionResult> Preconditions { get; }
```

Comment:
IMO naming this something like `PreconditionResults` instead would be more semantic. Additionally, it would probably make more sense as an IReadOnlyCollection rather than an IEnumerable.